sympy/solvers/inequalities.py

```diff
@@ -982,3 +985,487 @@ def reduce_inequalities(inequalities, symbols=[]):
+            eqns.append(rel.rhs - rel.lhs)
+            standard_constraints.append(-rel.lhs <= -rel.rhs)
+        elif type(rel) == Eq or isinstance(rel, AppliedBinaryRelation) and rel.function == Q.eq:
+            # x = 3 can be represented as x <= 3 and x >= 3
+            eqns.append(rel.lhs - rel.rhs)
+            standard_constraints.append(rel.lhs <= rel.rhs)
+            eqns.append(rel.rhs - rel.lhs)
+            standard_constraints.append(-rel.lhs <= -rel.rhs)
+        else:
+            raise TypeError(f"Unrecognized relation: {rel}")
+
+    A, B = linear_eq_to_matrix(eqns, *variables)
+    C = linear_eq_to_matrix(objective, *variables)[0] # constant terms can be safely ignored here
+
+    return A, B, C, standard_constraints
+
+
+def linprog_maximize_from_equations(constraints, objective, variables):
+    """
+    A function to maximize a linear objective function subject to linear
+    constraints with the simplex method. All coefficients must be Rational or
+    floats.
+
+    Parameters
+    ==========
+
+    constraints : list of linear inequalities
+        Stict inequalities (>, <) and not equals are not allowed.
+
+    objective : a linear function to maximize
+        If you want to minimize a function, f, simply make the objective
+        function -f.
+
+    variables : a list of variables included in constraints and objective
+
+    Returns
+    =======
+
+    Returns `(optimum, argmax, argmax_dual)`:
+
+    optimum : Rational or Float
+
+    argmax : dictionary of variables to Rationals or Floats
+
+    argmax_dual : dictionary of constraints to Rationals or Floats
+
+    Examples
+    ========
+
+    >>> from sympy.abc import x, y, z
+    >>> from sympy.solvers.inequalities import linprog_maximize_from_equations
+    >>> r1 = y+2*z <= 3
+    >>> r2 = -x-3*z <= -2
+    >>> r3 = 2*x+y+7*z <= 5
+    >>> optimum, argmax, argmax_dual  = linprog_maximize_from_equations([r1,r2,r3], x+y+5*z, [x, y, z])
+    >>> optimum
+    11/3
+    >>> argmax
+    {x: 0, y: 1/3, z: 2/3}
+    >>> argmax_dual
+    {-x - 3*z <= -2: 2/3, y + 2*z <= 3: 0, 2*x + y + 7*z <= 5: 1}
+
+    See Also
+    ========
+
+    solve_univariate_inequality
+    linprog_from_matrices
+    find_feasible
+    """
+    A, B, C, standard_constraints = _linear_programming_to_matrix(constraints, objective, variables)
+    optimum, argmax, argmax_dual = _simplex(A, B, C)
+    argmax = {variables[i] : argmax[i] for i in range(len(variables))}
+    argmax_dual = {standard_constraints[i] : argmax_dual[i] for i in range(len(standard_constraints))}
+    return optimum, argmax, argmax_dual
+
+
+def find_feasible(constraints, variables):
+    """
+    Finds a feasible solution to a system of linear inequalities, if any exist,
+    with the first phase of the simplex method.
+
+    Parameters
+    ==========
+
+    constraints : list of linear inequalities and equalities
+        Stict inequalities (>, <) and not equals are not allowed.
```

Comment:
I will implement handling for strict inequalities in find_feasible tomorrow. This should be fairly doable.

Basically, x < 0 can be represented as x + delta <= 0.